draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,28 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
+
 To define which function expressions are well typed,
 a type system is first introduced.
 
 ### Type System for Function Expressions {#typesys}
 
 Each parameter as well as the result of a function extension must have a declared type.
 
+The registered set of function extensions (see {{iana-fnex}}) is the source of knowledge
```

Comment:
This is maybe the place where we concede that an implementation may support further function extensions, which then are obviously not interoperable (and their use makes the queries using them non-interoperable).